OSDOCS Nodes Additional Artifact Store : GA 5.0 (#118494)
openshift/openshift-docs: modules/nodes-nodes-additional-crio-storage-about.adoc

When you create the container runtime config, the Machine Config Operator (MCO) writes the configuration to the `/etc/containers/storage.conf` file on the target nodes.

Additional container image layers for lazy pulling::
Use the `additionalLayerStores` field to enable lazy pulling through a third-party Bring Your Own Storage (BYOS) plugin. With lazy pulling, you can start a container without waiting for the entire image to be downloaded. Instead, the necessary parts of the image are fetched on-demand during runtime using FUSE.
+

> External BYOS plugins, such as stargz-snapshotter and nydus-storage-plugin, serve container image layers on-demand through a FUSE file system. The `additionalLayerStores` parameter configures the FUSE mount paths for CRI-O.

Reviewer: `additionalLayerStores` is called "field" on line 71, "parameter" here, and "object" on line 79. Three different terms in 8 lines. In OpenShift docs, "object" typically means a Kubernetes resource (a CR, a Pod), so using it for a CR field is confusing. Suggest "field" consistently.

```suggestion
External BYOS plugins, such as stargz-snapshotter and nydus-storage-plugin, serve container image layers on-demand through a FUSE file system. The `additionalLayerStores` field configures the FUSE mount paths for CRI-O.
```

+

> When CRI-O needs an image, it accesses the plugin's FUSE filesystem, triggering metadata download and lazy pulling. The container starts after downloading only required chunks.

Reviewer: "filesystem" (one word) here vs "file system" (two words) on line 73. Red Hat style prefers "file system" (two words).

```suggestion
When CRI-O needs an image, it accesses the plugin's FUSE file system, triggering metadata download and lazy pulling. The container starts after downloading only required chunks.
```

+
[NOTE]
====

> If you are using the Linux native zstd:chunked format for partial pulling, you do not need to configure a `ContainerRuntimeConfig` object. With partial pulling, CRI-O retrieves the chunk metadata, determines which chunks are needed, and fetches only those chunks using HTTP range requests rather than downloading the entire compressed layer.

Reviewer: This NOTE tells users they do not need a `ContainerRuntimeConfig`, but does not mention that `enable_partial_images` in `storage.conf` defaults to `"false"` in the MCO template, and there is no `ContainerRuntimeConfig` field to toggle it. Users would need a `MachineConfig` drop-in to set `enable_partial_images = "true"`.

As written, users will reasonably conclude zstd:chunked works with no configuration at all. Consider either mentioning the `MachineConfig` requirement or narrowing the statement to: "you do not need to configure `additionalLayerStores`" instead of "you do not need to configure a `ContainerRuntimeConfig` object."

Author: I made the update here.
But, is there a Jira that tracks the process for setting enable_partial_images to configure zstd:chunked, showing it was fully tested? The process really needs its own topic in the docs.

Reviewer: No, there seems to be no dedicated Jira for documentation on enable_partial_images. The broader zstd:chunked support is tracked under [OCPNODE-3137](https://redhat.atlassian.net/browse/OCPNODE-3137), but that epic is still "New".

Agreed it needs its own topic, since the configuration involves a MachineConfig drop-in, not a ContainerRuntimeConfig. That's out of scope for this PR. For now, I'd narrow this NOTE to say users don't need to configure additionalLayerStores for zstd:chunked, and I can file a story under [OCPNODE-3137](https://redhat.atlassian.net/browse/OCPNODE-3137) to track the dedicated zstd:chunked docs.

> If you are using the Linux native zstd:chunked format for partial pulling, you do not need to configure the `additionalLayerStores` object. With partial pulling, CRI-O retrieves the chunk metadata, determines which chunks are needed, and fetches only those chunks using HTTP range requests rather than downloading the entire compressed layer.

Reviewer: Same terminology issue: "object" → "field".

```suggestion
If you are using the Linux native zstd:chunked format for partial pulling, you do not need to configure the `additionalLayerStores` field. With partial pulling, CRI-O retrieves the chunk metadata, determines which chunks are needed, and fetches only those chunks using HTTP range requests rather than downloading the entire compressed layer.
```

====
+

> To use lazy pulling, you must install a BYOS plugin on your nodes. Then, create a container runtime config to configures the FUSE mount paths. 

Reviewer: Typo: "to configures" should be "to configure".

```suggestion
To use lazy pulling, you must install a BYOS plugin on your nodes. Then, create a container runtime config to configure the FUSE mount paths. 
```
